library/src/actions/jwt/jwt.ts

```diff
@@ -0,0 +1,193 @@
+import type {
+  BaseIssue,
+  BaseValidation,
+  ErrorMessage,
+} from '../../types/index.ts';
+import { _addIssue } from '../../utils/index.ts';
+
+const POSSIBLE_JWT_REGEX = /^(?:[\w-]+\.){2}[\w-]+$/u;
+
+// https://datatracker.ietf.org/doc/html/rfc7518#section-3.1
+type JwtAlgorithm =
+  | 'HS256'
+  | 'HS384'
+  | 'HS512'
+  | 'RS256'
+  | 'RS384'
+  | 'RS512'
+  | 'ES256'
+  | 'ES384'
+  | 'ES512'
+  | 'PS256'
+  | 'PS384'
+  | 'PS512'
+  | 'none';
+
+function isValidJwt(
+  input: string,
+  jwtAlgorithm: JwtAlgorithm | undefined
+): boolean {
+  if (!POSSIBLE_JWT_REGEX.test(input)) {
+    return false;
+  }
+  try {
+    // `atob`
+    // - for node env, marked as legacy
+    //     https://github.com/DefinitelyTyped/DefinitelyTyped/issues/65494
+    // - assumes the passed input to be base64 encoded,
+    //     will have to convert base64Url encoded header to base64
+    const header = JSON.parse(atob(input.split('.')[0]));
+    return (
+      'typ' in header &&
+      header.type === 'JWT' &&
+      (!('alg' in header) || header.alg === (jwtAlgorithm ?? 'none'))
+    );
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Jwt issue type.
+ */
+export interface JwtIssue<
+  TInput extends string,
+  TJwtAlgorithm extends JwtAlgorithm | undefined,
+> extends BaseIssue<TInput> {
+  /**
+   * The issue kind.
+   */
+  readonly kind: 'validation';
+  /**
+   * The issue type.
+   */
+  readonly type: 'jwt';
+  /**
+   * The expected property.
+   */
+  readonly expected: null;
+  /**
+   * The received property.
+   */
+  readonly received: `"${string}"`;
+  /**
+   * The algorithm used to sign the jwt.
+   */
+  readonly algorithm: TJwtAlgorithm;
+}
+
+/**
+ * Jwt action type.
+ */
+export interface JwtAction<
+  TInput extends string,
+  TJwtAlgorithm extends JwtAlgorithm | undefined,
+  TMessage extends ErrorMessage<JwtIssue<TInput, TJwtAlgorithm>> | undefined,
+> extends BaseValidation<TInput, TInput, JwtIssue<TInput, TJwtAlgorithm>> {
+  /**
+   * The action type.
+   */
+  readonly type: 'jwt';
+  /**
+   * The action reference.
+   */
+  readonly reference: typeof jwt;
+  /**
+   * The expected property.
+   */
+  readonly expects: null;
+  /**
+   * The algorithm used to sign the jwt.
+   */
+  readonly algorithm: TJwtAlgorithm;
```

Comment:
Same question mentioned [here](https://github.com/EltonLobo07/valibot/pull/1/files#r1807497874)

Reply:
Adding the algorithm parameter as a property to `JwtAction` looks good to me, but I would not add it to `JwtIssue` as issues so far only contain common properties. In simple words, we never added additional properties to issues in the past, but we added them to the action itself to make them accessible and allow them to be serialized and dezerialized sometime in the future.